carapace-server/src/main/java/org/carapaceproxy/client/impl/EndpointConnectionImpl.java

```diff
@@ -503,11 +450,11 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
 
     @Override
     public String toString() {
-        return "{cid=" + id + ", " + state + ", channel=" + channelToEndpoint + ", key=" + key + ", forcedInvalid=" + forcedInvalid + ", closed=" + closed + '}';
+        return "{cid=" + id + ", channel=" + channelToEndpoint + ", key=" + key + ", forcedInvalid=" + forcedInvalid + ", closed=" + closed + '}';
     }
 
     private void checkHandler(RequestHandler handler) throws IllegalStateException {
-        if (this.clientSidePeerHandler.get() != handler || !requestRunning) {
+        if (this.clientSidePeerHandler.get().getClientConnectionHandler().getId() != handler.getClientConnectionHandler().getId()) {
```

Comment:
Now same EndpointConnectionImpl may be used by same client for consecutive requests